vignettes/crunch.Rmd

```diff
@@ -0,0 +1,229 @@
+```
+
+The variables we included in the array have been converted into subvariables of the categorical array, and are still accessible using the `subvariables()` function.
+
+```{r}
+subvariables(ds$ImportantHiring)
+```
+
+
+#### Multiple Response
+
+Crunch also has a special variable type for Multiple Response questions. These are questions that give users a set of options and allow them to select more than one of them. An example of this from the Stack Overflow survey is asking which language a respondent wanted to work with. the responses are stored in a single variable with each language delimited by semi colons:
+
+```{r}
+knitr::kable(SO_survey[1:5, "HaveWorkedLanguage", drop = FALSE], row.names = FALSE)
+```
+
+To analyze this data without Crunch you would usually go through an elaborate data cleaning process to split the delimited variable up into its constituent parts, and then reshape it into a format which is more amenable to analysis. With Crunch you can make use of multiple response variables to store it in a compact form, and analyze it just like any other variable. Since delimited text is such a common way of storing this kind of answer, Crunch allows you to create a variable directly from the text. This function creates a _derived variable_ which means that if you make changes to the initial variable, the derived variable will change along with it.
+
+```{r state change5, echo=FALSE}
+change_state()
+```
+```{r makeMRFromText}
+ds$WantWorkedLanguageMR <- makeMRFromText(ds$WantWorkLanguage,
+    delim = "; ",
+    name = "WantWorkedMR")
+```
+```{r echo = FALSE}
+knitr::include_graphics("images/wantWorkMR.png")
+```
+
+The variable card now shows the number of times each language was selected, the sum of these numbers will be bigger than the number of respondents because people were allowed to select more than one language. We can reorder the subvariables to make this card easier to read.
+
+```{r reorder subvariables}
+counts <- as.array(crtabs(~ WantWorkedLanguageMR, ds))
+counts <- counts[rev(order(counts))]
+subvariables(ds$WantWorkedLanguageMR) <- subvariables(ds$WantWorkedLanguageMR)[names(counts)]
+```
+```{r echo = FALSE}
+knitr::include_graphics("images/WantWorkMRReodered.png")
+```
+
+## Archiving and deleting datasets
+
+Datasets can also be deleted permanently. This action cannot be undone, so it should not be done lightly. `crunch` provides two ways to delete a dataset: a `delete()` method on a dataset object, like
+
+```{r, eval=FALSE}
+## Not run
+delete(ds)
+```
+
+The second way to delete is `deleteDataset()`: you supply a dataset name. This way is faster if you have not already loaded the dataset object into your R session: no need to fetch something from the server just to then tell the server to delete it.
+
+```{r, include=FALSE}
+logout()
+end_vignette()
+```
+
+<!-- * [Datasets](datasets.html): creating, loading, and manipulating datasets in Crunch
```

Comment:
I think it would be good to have some sort of closing and keep some of this table of contents so that people know where to go next.